react/features/base/user-interaction/middleware.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import { APP_WILL_MOUNT, APP_WILL_UNMOUNT } from '../app';
+import { MiddlewareRegistry } from '../redux';
+
+import {
+    SET_USER_INTERACTION_LISTENER,
+    USER_INTERACTION_RECEIVED
+} from './actionTypes';
+
+/**
+ * Implements the entry point of the middleware of the feature base/user-interaction.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        _startListeningForUserInteraction(store);
+        break;
+
+    case APP_WILL_UNMOUNT:
+    case USER_INTERACTION_RECEIVED:
+        _stopListeningForUserInteraction(store);
+        break;
+    }
+
+    return next(action);
+});
+
+/**
+ * Registers listeners to notify redux of any user interaction with the page.
+ *
+ * @param {Object} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _startListeningForUserInteraction(store) {
+    const userInteractionListener = event => {
+        if (event.isTrusted) {
+            store.dispatch({
+                type: USER_INTERACTION_RECEIVED
+            });
+
+            _stopListeningForUserInteraction(store);
+        }
+    };
+
+    window.addEventListener('mousedown', userInteractionListener);
+    window.addEventListener('keydown', userInteractionListener);
+
+    store.dispatch({
```

Comment:
Do we need to store the listener in the store? It feels we can skip storing it there since we are using/removing it only from this function.

Reply:
makes sense

Reply:
To remove the listener then should we store it in the module? I guess we don't need to care about cleanup on any unmount for web.